app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -78,72 +76,51 @@ class TransferSendManager @Inject constructor(
     private suspend fun sendTransfer(uploadSessionUuid: String) {
         _sendStatus.value = SendStatus.Pending
 
-        withIntegrityToken(
-            onSuccess = { attestationToken ->
-                runCatching {
-                    uploadManager.initUploadSession(
-                        attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
-                        attestationToken = attestationToken,
-                    )!! // TODO: Handle ContainerErrorsException here
-                    uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
-                    _sendStatus.update {
-                        val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
-                        SendStatus.Success(totalSize)
-                    }
-                }.onFailure { exception ->
-                    SentryLog.e(TAG, "Failed to start the upload", exception)
-                    _sendStatus.update { SendStatus.Failure }
-                }
-            },
-            onRefused = { _sendStatus.update { SendStatus.Refused } },
-        )
+        runCatching {
+            val attestationToken = getAttestationToken()
+            uploadManager.initUploadSession(
+                attestationHeaderName = AppIntegrityManager.ATTESTATION_TOKEN_HEADER,
+                attestationToken = attestationToken,
+            )!! // TODO: Handle ContainerErrorsException here
+            uploadWorkerScheduler.scheduleWork(uploadSessionUuid)
+            _sendStatus.update {
+                val totalSize = importationFilesManager.importedFiles.value.sumOf { it.fileSize }
+                SendStatus.Success(totalSize)
+            }
+        }.onFailure { exception ->
+            if (exception !is NetworkException && exception !is KmpNetworkException) {
+                SentryLog.e(TAG, "Failed to start the upload", exception)
```

Comment:
I still think we need the tag to know which exceptions happen most of the time so we can priorized it and directly know which step of the process is causing problems